### ISSUE TITLE
Load relation ids without joining target entity

### ISSUE THREAD

**Issue by cgdangelo** [typeorm/typeorm#690]
_tl;dr what's the best way to fetch relation ids from a join table?_

I have a data model that is unfortunately spread out across several databases, so I need to handle some relations manually with separate queries on other connections. What I'm trying to do is find a way to fetch relation ids from a join table _only_; typeorm adds an additional join to the related class's table, which in my case, exists in another database.

```typescript
// Database A
@Entity('usr')
class User {
  @PrimaryGeneratedColumn()
  public id: number;

  @Column()
  public name: string;
}

// Database B
@Entity()
class Article {
  @PrimaryGeneratedColumn()
  public id: number;

  @ManyToMany(() => User)
  @JoinTable({name: 'article_authorship', inverseJoinColumn: {name: 'author_id'}})
  public authors: User[] = [];

  public authorIds: number[] = [];
}
```

The `article_authorship` join table is also on Database B and looks like:

```
   Column   |  Type   | Modifiers
------------+---------+-----------
 article_id | bigint  | not null
 author_id  | bigint  | not null
 ordinal    | integer | not null
```

I've tried a few different things in my controller to populate `Article#authors` or `Article#authorIds` with data, or otherwise obtain an array of ids from that join table.

### `SelectQueryBuilder#innerJoinAndMapMany`

```typescript
const articles = articleRepository.createQueryBuilder('article');

articles.innerJoinAndMapMany('article.authorIds', 'article_authorship', 'authorship', 'article.id = authorship.article_id');
```

This _almost_ works. The following query is executed:
```sql
SELECT "article"."id" AS "article_id", "article"."headline" AS "article_headline", "article"."content" AS "article_content", "authorship"."article_id" AS "authorship_article_id", "authorship"."author_id" AS "authorship_author_id" FROM "article" "article" INNER JOIN "article_authorship" "authorship" ON "article"."id" = authorship.article_id WHERE ("article"."date_deleted" IS NULL) AND "article"."id" IN ($1) -- PARAMETERS: ["41255"]
```

For an Article with 2 authors, this does in fact load 2 records with each User id; I can see that in the raw results from `SelectQueryBuilder#getRawAndEntities`. However, in the single entity that's returned, `Article#authorIds` is still empty. The transformation from raw results to entities leaves that out.

### `SelectQueryBuilder#loadAllRelationIds`

```typescript
const articles = articleRepository.createQueryBuilder('article');

articles.loadAllRelationIds();
```

This results in the following query:

```sql
SELECT "article_authors_relation_id"."author_id" AS "author_id", "article_authors_relation_id"."article_id" AS "article_id" FROM "usr" "usr" INNER JOIN "article_authorship" "article_authors_relation_id" ON ("article_authors_relation_id"."article_id" = $1 AND "article_authors_relation_id"."author_id" = "usr"."id") ORDER BY "article_authors_relation_id"."author_id" ASC, "article_authors_relation_id"."article_id" ASC -- PARAMETERS: ["41255"]
```

Obviously, this fails outright, as the `usr` table is on Database A. I also attempted to use `SelectQueryBuilder#loadRelationIdAndMap)` and heavily modify the query but no luck there.

**Comment by pleerock** [comment]
Do you simply want to load ids from junction (many-to-many) table, right?

```ts
    @ManyToMany(type => Post, post => post.categories)
    posts: Post[];

    @RelationId((category: Category) => category.posts)
    postIds: number[];
```

See examples [here](https://github.com/pleerock/typeorm/blob/cdae9fb31c707e54db63238fd3f8b2043dda682f/test/functional/decorators/relation-id/relation-id-many-to-many/relation-id-decorator-many-to-many.ts#L29-L29).

**Comment by cgdangelo** [comment]
Yeah, I just need to get the ids from the table. `@RelationId` also includes the join to the related entity's table (which is in a separate database):

```sql
SELECT "article_authors_relation_id"."author_id" AS "author_id", "article_authors_relation_id"."article_id" AS "article_id" FROM "usr" "usr" INNER JOIN "article_authorship" "article_authors_relation_id" ON ("article_authors_relation_id"."article_id" = $1 AND "article_authors_relation_id"."author_id" = "usr"."id") ORDER BY "article_authors_relation_id"."author_id" ASC, "article_authors_relation_id"."article_id" ASC -- PARAMETERS: ["41255"]
```

This results in `error: relation "usr" does not exist`, of course.

**Comment by pleerock** [comment]
wait, wait, did you say "different databases"? If yes then its not supported to do cross-databases queries, only requests in scope of single database is allowed

**Comment by pleerock** [comment]
@cgdangelo do you have something else to add here?

**Comment by pleerock** [comment]
Closing as no answer from reportee. Btw, cross-database requests are supported now. You can find docs [here](http://typeorm.io/#/multiple-connections).
